mlflow/utils/annotations.py

```diff
@@ -21,15 +22,23 @@ def deprecated(alternative=None, since=None):
     :param func: A function to mark
     :returns Decorated function.
     """
-    def deprecated_func(func):
-        since_str = " since %s" % since if since else ""
-        notice = ".. Warning:: Deprecated%s: This method will be removed in " % since_str + \
-                 "a near future release."
-        if alternative is not None and alternative.strip():
-            notice += " Use ``%s`` instead." % alternative
-        func.__doc__ = notice + "\n" + func.__doc__
-        return func
-    return deprecated_func
+
+    def deprecated_decorator(func):
+        @wraps(func)
+        def deprecated_func(*args, **kwargs):
+            since_str = " since %s" % since if since else ""
+            notice = ".. Warning::" + \
+                     func.__name__ + " is deprecated%s: This method will be removed in "\
+                     % since_str + "a near future release."
+            if alternative is not None and alternative.strip():
+                notice += " Use ``%s`` instead." % alternative
+            if func.__doc__ is not None:
+                func.__doc__ = notice + "\n" + func.__doc__
```

Comment:
Since we're wrapping `func`, this docstring change isn't being displayed in the docs (or in `help(mlflow.pyfunc.load_pyfunc)`):
![Screen Shot 2019-07-12 at 2 43 49 PM](https://user-images.githubusercontent.com/39497902/61160230-8d67c480-a4b3-11e9-9f30-50768e9b3ec0.png)

Reply:
This seems to work:
```
def deprecated(alternative=None, since=None):
    """
    Decorator for marking APIs deprecated in the docstring.

    :param func: A function to mark
    :returns Decorated function.
    """

    def deprecated_decorator(func):
        since_str = " since %s" % since if since else ""
        notice = (
            ".. Warning:: {function_name} is deprecated{since_string}. This method will be"
            " removed in a near future release".format(
                function_name=func.__name__,
                since_string=since_str)
        )
        if alternative is not None and alternative.strip():
            notice += " Use ``%s`` instead." % alternative

        @wraps(func)
        def deprecated_func(*args, **kwargs):
            warnings.warn(notice, category=DeprecationWarning, stacklevel=2)
            return func(*args, **kwargs)

        if func.__doc__ is not None:
            deprecated_func.__doc__ = notice + "\n" + func.__doc__

        return deprecated_func

    return deprecated_decorator
```

Reply:
Seems to work for me too, thanks @dbczumar